docs_src/microservices/device/V3Migration.md

```diff
@@ -8,29 +8,152 @@ See [Top Level V3 Migration Guide](../../../V3TopLevelMigration) for details app
 
 ### Device Files
 
-TBD
+1. Change device definition file in the Device Service to YAML format.
+2. Remove `LastConnected` and `LastReported` configs.
+3. ProtocolProperties now supports typed values.
+
+    !!! example "ProtocolProperty with typed values"
+    
+        ```yaml
+        protocols:
+          other:
+            Address: simple01
+            Port: 300
+        ```
 
 ### Device Profile Files
 
-TBD
+1. Add `optional` field in ResourceProperties to allow any additional or customized data.
+2. Change the data type of `mask`, `shift`, `scale`, `base`, `offset`, `maximum` and `minimum` from string to number in ResourceProperties.
+
 
 ### Provision Watcher files
 
-TBD
+1. Allow empty string profile name when adding or updating the ProvisionWatcher.
```

Comment:
Remove. This is not part of migrating an existing PW.

Reply:
removed